library/src/main/kotlin/build/buf/connect/Idempotency.kt

```diff
@@ -0,0 +1,51 @@
+// Copyright 2022-2023 Buf Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package build.buf.connect
+
+enum class Idempotency {
+    /**
+     * IdempotencyUnknown is the default idempotency level. A procedure with
+     * this idempotency level may not be idempotent. This is appropriate for
+     * any kind of procedure.
+     */
+    IDEMPOTENCY_UNKNOWN,
```

Comment:
Just `UNKNOWN`?

Reply:
Can. I was going off of: https://github.com/protocolbuffers/protobuf/blob/main/src/google/protobuf/descriptor.proto#L813C19-L817.

Reply:
Yeah I think we can bias towards language best practices in our interfaces